### PR TITLE
Fix luigi run use case

### PR DESCRIPTION
**Pull request by Tarrasch**
Please review @tym-oao :)

### CODE REVIEW

File: test/interface_test.py [hunk 1]
```diff
@@ -67,6 +70,19 @@ def test_interface_run_with_run_failure(self):
...
 
+    def test_just_run_main_task_cls(self):
+        class MyTestTask(luigi.Task):
+            pass
+
+        class MyOtherTestTask(luigi.Task):
+            my_param = luigi.Parameter()
+
+        with patch.object(sys, 'argv', ['my_module.py', '--no-lock', '--local-scheduler']):
+            luigi.run(main_task_cls=MyTestTask)
```
Comment (erikbern) [inline test/interface_test.py]: you can just pass `cmdline_args` to `luigi.run` instaed of patching sys.argv

**Comment by tym-oao** [comment]
Looks good! My original breaking case now works. Thanks for the fast attention!

**Comment by tym-oao** [comment]
Personally, I will still move to preferring `luigi` script over `main_task_cls`, but I appreciate not having to scramble to do that immediately all at once :)

**Comment by erikbern** [comment]
We could throw a deprecation warning to nudge people to move

File: test/interface_test.py (hunk 1 above)
Comment (Tarrasch) [inline test/interface_test.py]: Well yes, but the whole purpose is to test the behaviour when you don't pass that argument. As the commit message says. :)

**Comment by Tarrasch** [comment]
Erm, I don't know why. But apparently there are these 3 test faillures that always happen. I don't dare to merge this PR just yet. :-/

**Comment by Tarrasch** [comment]
Seem to be the same intermittent failures as I talked about here [1]. Merging to unbreak stuff for our beloved HEAD-users. :)

[1]: https://github.com/spotify/luigi/pull/1064#issuecomment-120597484

**Pull request merged**